tests/test_weighted_pandas.py

```diff
@@ -465,3 +413,181 @@ def test_WeightedSeries_nan():
     assert_allclose(series.mean(), 0.5, atol=1e-2)
     assert_allclose(series.var(), 1./12, atol=1e-2)
     assert_allclose(series.std(), (1./12)**0.5, atol=1e-2)
+
+
+@pytest.fixture
+def mcmc_df():
+    np.random.seed(0)
+    m, s = 0.5, 0.1
+
+    def logL(x):
+        return x, -((x-m)**2).sum(axis=-1)/2/s**2
+
+    np.random.seed(0)
+    x0, logL0 = logL(np.random.normal(m, s, 4))
+    dat = []
+    for _ in range(3000):
+        dat.append(x0)
+        x1, logL1 = logL(np.random.normal(x0, s/2))
+        if np.log(np.random.rand()) < logL1 - logL0:
+            x0, logL0 = x1, logL1
+
+    return DataFrame(dat, columns=["x", "y", "z", "w"])
+
+
+@pytest.fixture
+def mcmc_wdf(mcmc_df):
+    weights = mcmc_df.groupby(mcmc_df.columns.tolist(), sort=False).size()
+    return WeightedDataFrame(mcmc_df.drop_duplicates(), weights=weights.values)
+
+
+def test_WeightedDataFrame_hist(mcmc_df, mcmc_wdf):
+    df_axes = mcmc_df.hist()
+    wdf_axes = mcmc_wdf.hist()
+    for df_ax, wdf_ax in zip(df_axes.flatten(), wdf_axes.flatten()):
+        for df_patch, wdf_patch in zip(df_ax.patches, wdf_ax.patches):
+            assert df_patch.get_height() == wdf_patch.get_height()
+            assert df_patch.get_width() == wdf_patch.get_width()
+            assert df_patch.get_xy() == wdf_patch.get_xy()
```

Comment:
Instead of the `for` loops, here, it might be nicer to simply compare lists, e.g. for comparing heights:
```python
    df_axes = mcmc_df.hist().flatten()
    wdf_axes = mcmc_wdf.hist().flatten()
    df_heights = list(p.get_height() for ax in df_axes for p in ax.patches)
    wdf_heights = list(p.get_height() for ax in wdf_axes for p in ax.patches)
    assert df_heights == wdf_heights
```
In case of failure this will flag the list elements that are not equal.

Reply:
OK -- now does this [6e3f4b0](https://github.com/williamjameshandley/anesthetic/pull/194/commits/6e3f4b0993b46a24867ae93f3ca84ee8061c2b80). I agree this is much neater.